File: app/src/main/java/pl/piotrskiba/angularowo/MyFirebaseMessagingService.java
package pl.piotrskiba.angularowo;

import androidx.annotation.NonNull;

import com.google.firebase.messaging.FirebaseMessagingService;
import com.google.firebase.messaging.RemoteMessage;

import pl.piotrskiba.angularowo.utils.NotificationUtils;

public class MyFirebaseMessagingService extends FirebaseMessagingService {

    @Override
    public void onMessageReceived(@NonNull RemoteMessage remoteMessage) {
        super.onMessageReceived(remoteMessage);

        remoteMessage.getData();
        String notification_body = remoteMessage.getData().get(Constants.FIREBASE_FCM_DATA_NOTIFICATION_BODY);
        if(notification_body != null) {
            String notification_title = remoteMessage.getData().get(Constants.FIREBASE_FCM_DATA_NOTIFICATION_TITLE);

            boolean notification_sound = Boolean.parseBoolean(
                    remoteMessage.getData().get(Constants.FIREBASE_FCM_DATA_NOTIFICATION_SOUND)
            );

            new NotificationUtils(getApplicationContext())
                    .showNotification(notification_title, notification_body, notification_sound);
        }
    }
}

File: app/src/main/java/pl/piotrskiba/angularowo/MyFirebaseMessagingService.kt
package pl.piotrskiba.angularowo

import com.google.firebase.messaging.FirebaseMessagingService
import com.google.firebase.messaging.RemoteMessage
import pl.piotrskiba.angularowo.utils.NotificationUtils
import pl.piotrskiba.angularowo.utils.PreferenceUtils

class MyFirebaseMessagingService : FirebaseMessagingService() {
    override fun onMessageReceived(remoteMessage: RemoteMessage) {
        super.onMessageReceived(remoteMessage)
        val preferenceUtils = PreferenceUtils(applicationContext)

        // show notifications to logged in users only
        if (preferenceUtils.uuid != null) {
            val notificationBody = remoteMessage.data[Constants.FIREBASE_FCM_DATA_NOTIFICATION_BODY]

            notificationBody?.run {
                val notificationTitle = remoteMessage.data[Constants.FIREBASE_FCM_DATA_NOTIFICATION_TITLE]
                val notificationSound = remoteMessage.data[Constants.FIREBASE_FCM_DATA_NOTIFICATION_SOUND]?.toBoolean()

                if (notificationTitle != null && notificationSound != null) {
                    NotificationUtils(applicationContext)
                            .showNotification(notificationTitle, notificationBody, notificationSound)
                }
            }
        }
    }
}